### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by bikashamdocs**
@nagidocs, Please review the changes and let me know if any modification needed in this script.

### CODE REVIEW

File: Add_Credential_Jenkins.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+# Read each line from the input file and create a credential for each key-value pair
+while read line; do
+  key=$(echo $line | cut -d'=' -f1)
+  value=$(echo $line | cut -d'=' -f2)  
+  
+# Check if a credential with the same ID already exists
+  existing_credential=$(curl -s -o /dev/null -w "%{http_code}" -X GET -u "${USER_NAME}:${API_TOKEN}" "${JENKINS_URL}/credentials/store/system/domain/_/credential/${CREDENTIAL_ID}_${key}/api/json")
+  if [ ${existing_credential} -eq 200 ]; then
+    echo "Credential '${key}' already exists. Skipping."
```
Comment (nagidocs) [inline Add_Credential_Jenkins.sh:33]: Don't skip, instead update the existing with whatever we have in our file..we will mark it as a source of truth.

File: Add_Credential_Jenkins.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# This script reads key-value pairs from a file and creates a Jenkins credential
+
+# Replace the following variables with your Jenkins configuration
+JENKINS_URL="http://localhost:8080"
+CREDENTIAL_ID="your_credential_id"
```
Comment (nagidocs) [inline Add_Credential_Jenkins.sh:7]: Pass these 2 as well from command line arguments.

File: Add_Credential_Jenkins.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    --header "Content-Type:application/x-www-form-urlencoded" \
+    --fail >/dev/null 2>&1 # redirect curl output to null to avoid cluttering the terminal
+
+  # Check if the credential was created successfully
+  if [ $? -eq 0 ]; then
+    echo "Credential '${key}' created successfully."
+  else
+    echo "Failed to create credential '${key}'."
+  fi
+done < secrets.txt
```
Comment (nagidocs) [inline Add_Credential_Jenkins.sh:66]: Instead of fix secrets.txt, please make it flexible to pass file_path, where secret presents.

**Review by nagidocs — CHANGES_REQUESTED** [review]
Please make the necessary suggested changes.

**Comment by bikashamdocs** [comment]
Changes needed

**Pull request closed without merging**